pkcs8/src/common.rs

```diff
@@ -0,0 +1,76 @@
+use core::convert::TryFrom;
+
+use der::{Encodable, Encoder, Tagged};
+
+/// Version marker for PKCS#8 documents.
+///
+/// (RFC 5958 designates `0` and `1` as the only valid versions for PKCS#8 documents)
+#[derive(Clone, Debug, Copy)]
+pub enum Version {
+    /// Denotes PKCS#8 v1, used for [`crate::PrivateKeyInfo`] and [`crate::OneAsymmetricKey`]
+    V1 = 0,
+    /// Denotes PKCS#8 v2, only used for [`crate::OneAsymmetricKey`]
+    V2 = 1,
+}
+
+impl From<Version> for u8 {
+    fn from(version: Version) -> Self {
+        version as u8
+    }
+}
+
+impl TryFrom<der::Any<'_>> for Version {
+    type Error = der::Error;
+
+    fn try_from(any: der::Any<'_>) -> der::Result<Version> {
+        any.tag().assert_eq(Self::TAG)?;
+
+        match *any.as_bytes() {
+            [0x00] => Ok(Version::V1),
+            [0x01] => Ok(Version::V2),
+            _ => Err(der::ErrorKind::Noncanonical.into()),
```

Comment:
I think `der::ErrorKind::Value` would make more sense here:

```suggestion
            _ => Err(der::ErrorKind::Value { tag: der::Tag::Integer }),
```

The `Noncanonical` error kind is specifically for DER canonicalization errors (i.e. encodings which are valid under BER, but not under DER)